GitUI/UserControls/RevisionGrid/Columns/BuildStatusColumnProvider.cs

```diff
@@ -79,36 +79,15 @@ public override void OnCellPainting(DataGridViewCellPaintingEventArgs e, GitRevi
                 return;
             }
 
-            Size size;
+            string text = (AppSettings.ShowBuildStatusIconColumn ? revision.BuildStatus.StatusPicto : string.Empty)
+                + (AppSettings.ShowBuildStatusTextColumn ? (string)e.FormattedValue : string.Empty);
 
-            if (AppSettings.ShowBuildStatusIconColumn)
+            if (_fontWithUnicodeCache == null || _fontWithUnicodeCache.Size != style.NormalFont.Size)
             {
-                size = DpiUtil.Scale(new Size(8, 8));
-
-                Point location = new(
-                    e.CellBounds.Left + (size.Width / 2),
-                    e.CellBounds.Top + ((e.CellBounds.Height - size.Height) / 2));
-
-                GraphicsContainer container = e.Graphics.BeginContainer();
-                e.Graphics.SmoothingMode = SmoothingMode.AntiAlias;
-                using Brush brush = CreateCircleBrush();
-                e.Graphics.FillEllipse(brush, new Rectangle(location, size));
-                e.Graphics.EndContainer(container);
-            }
-            else
-            {
-                size = default;
+                _fontWithUnicodeCache = new Font(FontFamily.GenericMonospace, style.NormalFont.Size);
```

Comment:
Can we pre-allocate or allocate only once?

Reply:
It is already the case. We allocate only once or when the font size change (which should be not that often)